### PR TITLE
Environments docs

### PR DESCRIPTION
**Pull request by jessewgibbs**

### CODE REVIEW

**Comment by jessewgibbs** [comment]
This docs page has been on my back burner for a while, but I wanted to get some ideas written down and circulated.  Looking forward to getting some feedback from the team on these ideas & suggested best practices.

File: guides/environments.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Environments
+
+Every error reported to Rollbar is required to have an `environment` property.  This value can contain any non-empty alphanumeric string of up to 255 characters, so it can be used in a number of ways.
```
Comment (brianr) [inline guides/environments.md]: - Technically it's not actually required. If you omit environment, it'll become the string `"unknown"`
- Can contain "any non-empty string of up to 255 latin-1 characters"

File: guides/environments.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+This guide explains a bit about how `environment` is used in the app, some important limitations, and our recommended best practices.
+
+## Important things to know about `environment` in Rollbar:
+
+* `production` is treated as the default environment filter in many Rollbar screens, since we assume that production data matters the most.
+* Items cannot be [merged](/docs/merging/) if they are in different environments.
+* [Notification rules](/docs/notifications/) can easily be applied to a single environment or all environmnts, but not necessarily to multiple environments.
```
Comment (brianr) [inline guides/environments.md]: - typo: `all environmnts`

File: guides/environments.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+## Best practices
+
+* In general, we recommend to have a single `production` environment to enable merging and accurate metrics for all errors that occur in your production systems.
+  * For development builds, many customers use the git branch name or the sandbox name as `environment`.  We recommend this approach.
+* To track which host an error occurred on, use the `host` property when reporting occurrences (see the docs for your [Rollbar SDK](/docs/notifiers/) for instructions on sending the `host` property.)
+  * If all hosts are in the same location, then a `host` format like `web1` is recommended.
+  * If hosts are spread across multiple regions / data centers, a format like `us-east1-web1 ` is recommended.
```
Comment (brianr) [inline guides/environments.md]: Worth mentioning that when you search by host, it's a prefix search - so if you do it like this, then you can do `host:us-east1-` to find everything on all of the us-east1 hosts.

File: guides/environments.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+## Important things to know about `environment` in Rollbar:
+
+* `production` is treated as the default environment filter in many Rollbar screens, since we assume that production data matters the most.
+* Items cannot be [merged](/docs/merging/) if they are in different environments.
+* [Notification rules](/docs/notifications/) can easily be applied to a single environment or all environments, but not necessarily to multiple environments.
+
+## Best practices
+
+* In general, we recommend to have a single `production` environment to enable merging and accurate metrics for all errors that occur in your production systems.
+  * For development builds, many customers use the git branch name or the sandbox name as `environment`.  We recommend this approach.
```
Comment (rivkahstandig3636) [inline guides/environments.md]: Spacing is off here

File: guides/environments.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+## Important things to know about `environment` in Rollbar:
+
+* `production` is treated as the default environment filter in many Rollbar screens, since we assume that production data matters the most.
+* Items cannot be [merged](/docs/merging/) if they are in different environments.
+* [Notification rules](/docs/notifications/) can easily be applied to a single environment or all environments, but not necessarily to multiple environments.
+
+## Best practices
+
+* In general, we recommend to have a single `production` environment to enable merging and accurate metrics for all errors that occur in your production systems.
```
Comment (rivkahstandig3636) [inline guides/environments.md]: Nitpicky, but I would say "we recommend having"

File: guides/environments.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  * For development builds, many customers use the git branch name or the sandbox name as `environment`.  We recommend this approach.
+* To track which host an error occurred on, use the `host` property when reporting occurrences (see the docs for your [Rollbar SDK](/docs/notifiers/) for instructions on sending the `host` property.)
```
Comment (rivkahstandig3636) [inline guides/environments.md]: I would put the period outside of the parenthesis.

**Pull request closed without merging**